src/core/pointcloud/qgseptdecoder.cpp

```diff
@@ -466,4 +478,24 @@ QgsPointCloudBlock *QgsEptDecoder::decompressLaz( const QString &filename,
          );
 }
 
+QgsPointCloudBlock *QgsEptDecoder::decompressLaz( const QByteArray &byteArrayData,
+    const QString &nodeStr,
+    const QgsPointCloudAttributeCollection &attributes,
+    const QgsPointCloudAttributeCollection &requestedAttributes )
+{
+  QString filename = QDir::tempPath() + QDir::separator() + QStringLiteral( "%1.laz" ).arg( nodeStr );
```

Comment:
naming files like this is not unique enough... consider 1. there could be multiple point cloud layers loaded at once, 2. there could be multiple QGIS instances running.

A good idea may be to use QTemporaryFile which guarantees safe naming of temporary files - and has functionality to auto-remove the temp files.